### ISSUE TITLE
Getting the crashplan memory consumption error - How do I increase the memory?

### ISSUE THREAD

**Issue by bgmess** [jlesage/docker-crashplan-pro#105]
I have 6GB of memory in my Synology - I need instructions on how to increase this value.  I am fairly new to Linux and am not sure.  I read some of the other posts here regarding the error and got more confused.  I understand I should start at 4.5 GB and go down from there.....but how do I set that value?  You have been most helpful in the past - so thanks in advance.

Brian

**Comment by jlesage** [comment]
You basically have to change the value of the `CRASHPLAN_SRV_MAX_MEM` variable.  This can be done in the container's settings.
See https://github.com/jlesage/docker-crashplan-pro#environment-variables for more details about this variable

**Comment by bigtfromaz** [comment]
Is this folder, 
/volume1/docker/appdata/crashplan-pro
selected for backup in the Crashplan app?

**Comment by jlesage** [comment]
> Is my syntax above correct:  -e crashplan_serv_max_mem=5.0G

The name of the variable must be in uppercase and you have a typo:  It should be `CRASHPLAN_SRV_MAX_MEM=5G`

**Comment by bigtfromaz** [comment]
Also you might try adding some memory parameters to your docker run command.  I don't know what the defaults are but I'll bet it's lower than 5GB.  I assume you have that much free memory on you Synology.

There are a number of memory related parameters you might want to research.  Here is a documentation link for [docker run](https://docs.docker.com/engine/reference/commandline/run/).

**Comment by bgmess** [comment]
Yes, it is.

 

Should I deselect it?

 

Brian

**Comment by bigtfromaz** [comment]
There is no need to select that folder.  I worry that it could send you into a loop where crashplan activity triggers a backup which creates activity, and so on...

Code42 may well detect this and break the loop but why flirt with trouble for no good reason?

From my mobile device. Please forgive me if the message seems terse.

________________________________

**Comment by wheder** [comment]
> You basically have to change the value of the `CRASHPLAN_SRV_MAX_MEM` variable. This can be done in the container's settings.
> See https://github.com/jlesage/docker-crashplan-pro#environment-variables for more details about this variable

I don't believe it actually works on 100% - after tons of tries, I really had to add `-Xmx8192m` to the `SRV_JAVA_OPTS` in `bin/run.conf`, otherwise the java process was killed at cca 25% of the total system memory.

**Comment by jlesage** [comment]
Did you check the content of `bin/run.conf` when the environment variable was used?

**Comment by wheder** [comment]
Yes, it wasn't there initially. Or should I remove that file after increasing `CRASHPLAN_SRV_MAX_MEM`?

**Comment by jlesage** [comment]
Normally you don't need to manually edit this file.  It is automatically installed during the first start or during an upgrade.  To restore the default one, you can do:
```
docker exec [container name] cp /defaults/run.conf /config/bin/run.conf
```
When `CRASHPLAN_SRV_MAX_MEM` is set, `run.conf` is automatically modified (during container start) to change the value of the `-Xmx` parameter.

**Comment by wheder** [comment]
I have never seen the `-Xmx` in the `run.conf`, until added manually :-(

I also do not see the value (`-Xmx8192m`) changed after container restart (`stop`, `rm`, `run`), as I am having `CRASHPLAN_SRV_MAX_MEM=11G`

**Comment by jlesage** [comment]
If `CRASHPLAN_SRV_MAX_MEM` is currently set, try to restore (while the container is running) the default `run.conf` file.  Then restart the container (`docker restart`).

For sure you should see the `-Xmx` parameter, since it is present inside the default config file.  If however it is not modified automatically, please provide the output of `docker inspect [container name]`.

**Comment by wheder** [comment]
Unbelievable! That worked, thank you.

**Comment by prajitr81** [comment]
I've set the CRASHPLAN_SRV_MAX_MEM to 15G on my Synology DS1618+ with 16GB ram. The backup set is 5 TB. The app still keeps crashing due to memory issues. It worked fine during restore.

**Comment by jlesage** [comment]
Can you check `/config/bin/run.conf` to make sure the `-Xmx` parameter is properly set?

**Comment by prajitr81** [comment]
I am unable to locate that file. I don't see a crashplan directory under /volume1/docker

**Comment by jlesage** [comment]
You should be able to find `bin/run.conf` under the directory you mapped to `/config`.

**Comment by prajitr81** [comment]
This is all I see in that directory. 
<img width="423" alt="Screen Shot 2020-03-20 at 9 52 53 PM" src="https://user-images.githubusercontent.com/28905495/77218832-3bbd1280-6af5-11ea-896e-fd0647ddace1.png">

Thanks for your responses so far.

**Comment by prajitr81** [comment]
I think I found a workaround. I modified the file jvm_args under /volume1/docker/conf. It had only one entry for Xmx=1024. Not sure why I didn't see the run.conf file.

**Comment by jlesage** [comment]
Humm thanks I will look at this... Things may have changed in latest version of CP...

**Comment by jlesage** [comment]
Note that this is now fixed in latest image.
